sklearn/utils/_array_api.py

```diff
@@ -560,20 +569,50 @@ def get_namespace(*arrays, remove_none=True, remove_types=(str,), xp=None):
     return namespace, is_array_api_compliant
 
 
-def get_namespace_and_device(*array_list, remove_none=True, remove_types=(str,)):
-    """Combination into one single function of `get_namespace` and `device`."""
-    array_list = _remove_non_arrays(
-        *array_list, remove_none=remove_none, remove_types=remove_types
+def get_namespace_and_device(*arrays, remove_none=True, remove_types=(str,), xp=None):
+    """Combination into one single function of `get_namespace` and `device`.
+
+    Parameters
+    ----------
+    *arrays : array objects
+        Array objects.
+
+    remove_none : bool, default=True
+        Whether to ignore None objects passed in arrays.
+
+    remove_types : tuple or list, default=(str,)
+        Types to ignore in the arrays.
+
+    xp : module, default=None
+        Precomputed array namespace module. When passed, typically from a caller
+        that has already performed inspection of its own inputs, skips array
+        namespace inspection.
+
+    Returns
+    -------
+    namespace : module
+        Namespace shared by array objects. If any of the `arrays` are not arrays,
+        the namespace defaults to NumPy.
+
+    is_array_api_compliant : bool
+        True if the arrays are containers that implement the Array API spec.
+        Always False when array_api_dispatch=False.
+
+    device : device
+        `device` object (see the "Device Support" section of the array API spec).
+    """
```

Comment:
Note to reviewers: for some reason, the numpy docstring test started to complain about the lack of params / returns in this docstring only on this PR instead of the PR that introduced it.

I am not sure why but I decided to fix it here. We can take it apart in a dedicated PR if needed.